### PR TITLE
Add qos config and buffer profile

### PR DESCRIPTION
**Pull request by ysmanman**
Add qos config and buffer profile for 7800R3 48CQ2 Linecard.

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106

### CODE REVIEW

File: device/arista/x86_64-arista_7800r3_48cq2_lc/Arista-7800R3-48CQ2-C48/buffers.json.j2 [hunk 1]
```diff
@@ -0,0 +1,2 @@
+{%- set default_topo = 't0' %}
```
Comment (smaheshm) [inline device/arista/x86_64-arista_7800r3_48cq2_lc/Arista-7800R3-48CQ2-C48/buffers.json.j2:1]: 't2'?

**Comment by smaheshm** [comment]
@vmittal-msft Can you check the buffer profile values and comment if needs any updates.

**Comment by smaheshm** [comment]
@ysmanman Can you update the PR with suggestions. We can update the threshold values in a separate PR.

File: device/arista/x86_64-arista_7800r3_48cq2_lc/Arista-7800R3-48CQ2-C48/buffers.json.j2 (hunk 1 above)
Comment (ysmanman) [reply device/arista/x86_64-arista_7800r3_48cq2_lc/Arista-7800R3-48CQ2-C48/buffers.json.j2:1]: @smaheshm Thanks for catching this. Will fix.

Comment (vmittal-msft) [reply device/arista/x86_64-arista_7800r3_48cq2_lc/Arista-7800R3-48CQ2-C48/buffers.json.j2:1]: @ysmanman , Please let us know once done. Planning to start verification for these settings.

Comment (ysmanman) [reply device/arista/x86_64-arista_7800r3_48cq2_lc/Arista-7800R3-48CQ2-C48/buffers.json.j2]: @vmittal-msft Updated the review to address the above comment. We were testing PFC with a SAI patch from Broadcom ( Broardcom case: CS00012207728).  You may need this patch as well if you want test PFC. Add @rlhui for viz.

Comment (vmittal-msft) [reply device/arista/x86_64-arista_7800r3_48cq2_lc/Arista-7800R3-48CQ2-C48/buffers.json.j2]: @ysmanman Thanks for quick response. I am unable to access CS00012207728. Is it possible for you to share some detail ?

Thanks
Vineet

Comment (ysmanman) [reply device/arista/x86_64-arista_7800r3_48cq2_lc/Arista-7800R3-48CQ2-C48/buffers.json.j2]: @vmittal-msft We uncovered some issues in PFC testing, e.g., switch didn't send pause frame to to upstream even it received pause frames from downstream in certain scenarios; and storm mitigation was not working (switch kepts sending pause frame even it's in stormed state) when PFCWD was enabled. We filed few Broadcom cases, CS00012203593 and CS00012207728. Broadcom provides a patch (on top of SAI 5.2) for above issues, and we have been testing the patch recently.

**Review by vmittal-msft — APPROVED** [review]
Looks good to me. We will update these settings based on our usage scenarios.

**Pull request merged**